### ISSUE TITLE
执行fluid.optimizer.AdamOptimizer.minimize()函数报错

### ISSUE THREAD

**Issue by shaofeifei11** [PaddlePaddle/Paddle#26456]
PaddlePaddle版本：1.8.1
 
center_loss = fluid.layers.sigmoid_cross_entropy_with_logits(label=self.center_ness, x=self.predict_center)
self.center_loss = fluid.layers.reduce_mean(center_loss, name="center_loss")

self.opt = fluid.optimizer.AdamOptimizer(learning_rate=0.001, beta1=0.9, beta2=0.999)
self.opt.minimize(self.center_loss)  #  这一段报错

报错信息：
![image](https://user-images.githubusercontent.com/50863459/90620246-c80a6780-e244-11ea-84fb-12dab789fd63.png)

新手入门，感谢大佬指导

**Comment by hong19860320** [comment]
能把完整的脚本发下吗？

**Comment by shaofeifei11** [comment]
谢谢，我已经复现问题了。我简写了一个例子，找到相同问题了。

错误代码：
from paddle import fluid
import numpy as np

x = fluid.layers.data(shape=[-1, 3], dtype='float32', name="x", append_batch_size=False)

w = fluid.layers.create_parameter(shape=[3, 2], dtype="float32", name="w",
                                  attr=fluid.ParamAttr(name="w", regularizer=fluid.regularizer.L2Decay))
                ### 如果 fluid.regularizer.L2Decay 不加（）的话，就会报错，

mid = fluid.layers.matmul(x, w)
loss = fluid.layers.mean(mid)

opt = fluid.optimizer.AdamOptimizer(learning_rate=0.001, beta1=0.9, beta2=0.999)
opt.minimize(loss)

place = fluid.CPUPlace()
exe = fluid.Executor(place)
start_program = fluid.default_startup_program()
exe.run(start_program)

for i in range(10):
    lo = exe.run(feed={"x": np.asarray([[1,2,3],
                             [4,5,6]], dtype="float32")}, fetch_list=[loss])
    print(lo)

报错内容：
![image](https://user-images.githubusercontent.com/50863459/90714116-88418f80-e2d9-11ea-9e75-b2b73980e662.png)


正确代码：
from paddle import fluid
import numpy as np

x = fluid.layers.data(shape=[-1, 3], dtype='float32', name="x", append_batch_size=False)

w = fluid.layers.create_parameter(shape=[3, 2], dtype="float32", name="w",
                                  attr=fluid.ParamAttr(name="w", regularizer=fluid.regularizer.L2Decay()))
                ### 如果 fluid.regularizer.L2Decay 不加（）的话，就会报错，

mid = fluid.layers.matmul(x, w)
loss = fluid.layers.mean(mid)

opt = fluid.optimizer.AdamOptimizer(learning_rate=0.001, beta1=0.9, beta2=0.999)
opt.minimize(loss)

place = fluid.CPUPlace()
exe = fluid.Executor(place)
start_program = fluid.default_startup_program()
exe.run(start_program)

for i in range(10):
    lo = exe.run(feed={"x": np.asarray([[1,2,3],
                             [4,5,6]], dtype="float32")}, fetch_list=[loss])
    print(lo)

**Comment by hong19860320** [comment]
赞👍🏻 这个错误很隐蔽，看样子fluid.ParamAttr需要加参数check
